### PR TITLE
Transport Officer

### PR DESCRIPTION
**Pull request by Barnet2**
## About The Pull Request
Reduces PO slot to one, adding a new role: Transport Officer
25 hour timelock (pending other PR), has engi skill instead of pilot skill
TO comes with some materials in a construction pouch (50 metal, 50 plasteel, 5 sandbags and 20 barb) alongside a welderpack
Pilot Officer by default can no longer pilot tad
Pilot Officer tablet only sees Alamo, while TO sees tadpole. Tadpole has actual working cameras now
## Why It's Good For The Game
Attempts to fix two issues:
1. PO's arguing over who gets what (condor/tad)
2. New players picking PO and griefing
Number one is self explanatory. Number two should be somewhat remedied (but not completely fixed due to the nature of the tadpole) with a high timelock.
## Changelog
:cl:
add: New role: Transport Officer
balance: Reduces PO slot to one
fix: Tadpole cameras use the proper network
/:cl:

### CODE REVIEW

**Comment by Al-1ce** [comment]
This has been tried a dozen times already and for whatever reason it always gets shot down.

**Comment by efzapa** [comment]
> This has been tried a dozen times already and for whatever reason it always gets shot down.

mostly due to people coping about their precious playtime being lost

**Comment by novaepee** [comment]
we should call PO CAS pilot if this pr gets merged

**Comment by Loafin34** [comment]
sovl removed.. also job bloat.

**Comment by spookydonut** [comment]
this is dumb

**Comment by FosterGrime-Heymynameslouie** [comment]
this is really good I dont understand why people dislike it
basically creates distinction between cas and po so they dont have to do a fist fight between who gets what when they have conflicting intrests
obliterates grief, seriously why does po in general has such little timelock?
spedmgehr

**Comment by Woker2004** [comment]
Barnet I will state, please rename old PO to CAS pilot to avoid confusion, other than that I can agree with this change

**Comment by AusarAtkin** [comment]
Do other roles can still pilot Tad with access on their id, or its going to be like a mech and its pilot?

**Comment by 1kajowsmol** [comment]
needs tm for like.. several days to really get a feel. seems good idea tho

**Comment by Woker2004** [comment]
> This cannot be done due to how persistence works based off of job names. It would kill existing PO hours.

There are like only one person who cares about PO hours, Cassie is the only one who really would care

**Comment by Barnet2** [comment]
> > This cannot be done due to how persistence works based off of job names. It would kill existing PO hours.
> 
> 
> 
> There are like only one person who cares about PO hours, Cassie is the only one who really would care

That does not matter, maintainers want to keep the hours so PO will stay named PO, no exceptions.

**Comment by Woker2004** [comment]
> That does not matter, maintainers want to keep the hours so PO will stay named PO, no exceptions.
Maints back at it with another 1984

File: code/game/objects/effects/landmarks/marine_spawns.dm [hunk 1]
```diff
@@ -34,6 +34,10 @@
 	icon_state = "PO"
 	job = /datum/job/terragov/command/pilot
 
+/obj/effect/landmark/start/job/tadofficer
+	icon_state = "PO"
+	job = /datum/job/terragov/command/tadofficer
+
```
Comment (TiviPlus) [inline code/game/objects/effects/landmarks/marine_spawns.dm:40]: should probably use a different icon, no?

File: code/game/objects/items/devices/tablets.dm [hunk 2]
```diff
@@ -63,6 +63,10 @@
 				dat += " pilot's"
 				network = list("dropship1", "dropship2")
 				req_access = list(ACCESS_MARINE_PILOT, ACCESS_MARINE_DROPSHIP)
+			if(/datum/job/terragov/command/tadofficer)
+				dat += " pilot's"
+				network = list("dropship1", "dropship2")
+				req_access = list(ACCESS_MARINE_PILOT, ACCESS_MARINE_TADPOLE)
```
Comment (TiviPlus) [inline code/game/objects/items/devices/tablets.dm:69]: copy paste dat, network codes are still shared despite job being split

File: code/__DEFINES/jobs.dm [hunk 3]
```diff
@@ -147,6 +149,7 @@ GLOBAL_LIST_INIT(jobs_fallen_marine, typecacheof(list(/datum/job/fallen/marine),
 #define XP_REQ_INTERMEDIATE 180
 #define XP_REQ_EXPERIENCED 600
 #define XP_REQ_EXPERT 900
+#define XP_REQ_TADPOLE_OFFICER 3000
```
Comment (TiviPlus) [inline code/__DEFINES/jobs.dm:152]: Remove the new define and set it to expert, we will increase the define after to 20/25h

This was agreed on in maintainer chat

File: code/__DEFINES/jobs.dm [hunk 4]
```diff
@@ -47,6 +48,7 @@
 #define EXECUTIVE_OFFICER "Executive Officer" //Currently disabled.
 #define FIELD_COMMANDER "Field Commander"
 #define STAFF_OFFICER "Staff Officer"
+#define TADPOLE_OFFICER "Tadpole Officer"
```
Comment (TiviPlus) [inline code/__DEFINES/jobs.dm:51]: This name is not new player friendly (new player thinks tadpole is an animal), rename it to something like transport pilot
Might want to rename rest of code for consistency as well

File: code/datums/jobs/job/shipside.dm [hunk 5]
```diff
@@ -291,12 +291,65 @@ You are in charge of logistics and the overwatch system. You are also in line to
...
+			new_human.wear_id.paygrade = "CWO"
+		if(3001 to 6000) // 50 hrs
+			new_human.wear_id.paygrade = "O1"
+		if(6001 to INFINITY) // 100 hrs
+			new_human.wear_id.paygrade = "O2"
+
+/datum/job/terragov/command/tadofficer/radio_help_message(mob/M)
+	. = ..()
+	to_chat(M, {"Your job is to support marines mobile dropship support with the Tadpole.
+You are to ensure the Tadpole's survival and to transport marines around, acting as a mobile bunker. In the case of it's death, you are relegated to the FOB and the Alamo.
```
Comment (TiviPlus) [inline code/datums/jobs/job/shipside.dm:344]: Is this official admin policy? Last I checked currently if tad dies the PO can go do whatever. Get me a ruling from headmins on this please

File: code/datums/jobs/job/shipside.dm [hunk 6]
```diff
@@ -316,7 +369,7 @@ You are in charge of logistics and the overwatch system. You are also in line to
 		<b>You answer to the</b> acting Command Staff<br /><br />
 		<b>Unlock Requirement</b>: Starting Role<br /><br />
 		<b>Gamemode Availability</b>: Nuclear War<br /><br /><br />
-		<b>Duty</b>: Choose between the Condor, a modular attack aircraft that provides close air support with a variety of weapons ranging from the inbuilt gatling to wing mounted rockets; or the Tadpole, a versatile dropship capable of fulfilling roles ranging from ambulance to mobile bunker.
+		<b>Duty</b>: Pilot the Condor, a modular attack aircraft that provides close air support with a variety of weapons ranging from the inbuilt gatling to wing mounted rockets; or the Tadpole, a versatile dropship capable of fulfilling roles ranging from ambulance to mobile bunker.
```
Comment (TiviPlus) [inline code/datums/jobs/job/shipside.dm:372]: Changed the choose but didnt remove the option lol

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest lgtm

**Comment by TiviPlus** [comment]
> > > This cannot be done due to how persistence works based off of job names. It would kill existing PO hours.
> > 
> > 
> > There are like only one person who cares about PO hours, Cassie is the only one who really would care
> 
> That does not matter, maintainers want to keep the hours so PO will stay named PO, no exceptions.

Like I said in discord this also ties to job bans so this is an admin issue

**Comment by TiviPlus** [comment]
> > Do other roles can still pilot Tad with access on their id, or its going to be like a mech and its pilot?
> 
> They just need the access, yes. The only ones with tad access by default are the TO, FC and SLs.

I don't see you adding tad access to SLs here?

**Comment by breadhunt** [comment]
> this is dumb

hard disagree, other than what foster already said this makes tad PO/cas PO significantly easier to balance (no more CSE/synth as PO+ for tad), streamlines pre-drop ("hey do we have a tad po? ok lemme check manifest")
there are a lot of benefits from this which make it worth adding -- significantly more so i would argue than roles like ST or CL

**Review by Kuro020 — APPROVED** [review]

**Comment by chizzy376** [comment]
> This has been tried a dozen times already and for whatever reason it always gets shot down.

Because walancers hate the idea of a PO officer being self sustainable
Look at Derrick's CSE jobban for example 😂

**Comment by bowlofcereal** [comment]
> > This has been tried a dozen times already and for whatever reason it always gets shot down.
> 
> Because walancers hate the idea of a PO officer being self sustainable Look at Derrick's CSE jobban for example 😂
> 
> This is a good PR and the only reason you don't want it to get merged is because you HATE good change

what

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**